src/crimson/os/seastore/extent_placement_manager.cc

```diff
@@ -273,7 +277,9 @@ SegmentedAllocator::Writer::roll_segment(bool set_rolling) {
     });
   }
 
-  return segment_provider.get_segment().safe_then([this](auto segment) {
+  return segment_provider.get_segment(
+    segment_manager.get_device_id()
```

Comment:
Does this allocator actually need a SegmentManager instance?  Can we just give it a device id?

Reply:
Um, segment manager is also used to open/close segments in the writer

Reply:
Ah, right.